modules/apps/frontend-token/frontend-token-definition-impl/src/main/java/com/liferay/frontend/token/definition/internal/ThemeBundleTrackerCustomizer.java

```diff
@@ -0,0 +1,181 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.frontend.token.definition.internal;
+
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.model.PortletConstants;
+import com.liferay.portal.kernel.resource.bundle.ResourceBundleLoader;
+import com.liferay.portal.kernel.resource.bundle.ResourceBundleLoaderUtil;
+import com.liferay.portal.kernel.util.Portal;
+import com.liferay.portal.kernel.util.URLUtil;
+
+import java.io.IOException;
+
+import java.net.URL;
+
+import java.util.Dictionary;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleEvent;
+import org.osgi.util.tracker.BundleTrackerCustomizer;
+
+/**
+ * @author Anderson Luiz
+ * @author Thiago Buarque
+ */
+public class ThemeBundleTrackerCustomizer
+	implements BundleTrackerCustomizer<FrontendTokenDefinitionImpl> {
+
+	public ThemeBundleTrackerCustomizer(
+		FrontendTokenDefinitionManager manager, Portal portal) {
+
+		_manager = manager;
+		_portal = portal;
+	}
+
+	@Override
+	public FrontendTokenDefinitionImpl addingBundle(
+		Bundle bundle, BundleEvent event) {
+
+		String themeId = null;
+
+		try {
+			String frontendTokenDefinitionAsJsonString =
+				_getFrontendTokenDefinitionJSONFromBundle(bundle);
+
+			if (Objects.nonNull(frontendTokenDefinitionAsJsonString)) {
+				themeId = _getThemeId(bundle, _portal);
```

Comment:
Why are we passing _portal into _getThemeId when _portal is an instance variable of the class?

Reply:
oh you're right, I'm missed it. good catch